src/core/graph.js

```diff
@@ -7,6 +7,80 @@ import {makeAddressModule, type AddressModule} from "./address";
 import {toCompat, fromCompat, type Compatible} from "../util/compat";
 import * as NullUtil from "../util/null";
 
+/**
+ * This module contains the Graph, which is one of the most fundamental pieces
+ * of SourceCred. SourceCred uses this graph to model all of the contributions
+ * that make up a project, and the relationships between those contributions.
+ *
+ * The Graph serves a simple function: it keeps track of which Nodes exist, and
+ * what Edges join those nodes to each other. Nodes and Edges are both identified
+ * by Addresses; specifically, `NodeAddressT`s and `EdgeAddressT`s.
```

Comment:
Here we may want to add a link to a resource introducing the concept of graphs more generally. This seems like a good candidate: https://softwareengineering.stackexchange.com/questions/168058/what-are-graphs-in-laymens-terms#168067

Reply:
Done, thanks